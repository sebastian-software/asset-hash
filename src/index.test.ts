/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-call */

// eslint-disable-next-line shopify-lean/no-ancestor-directory-import
import { Hasher, getHash, getHashedName } from "."

const testString =
  "A stream constructor that takes in the seed to use. "
  + "Write data to the stream and when the stream ends."

test("Encode text", async () => {
  const hash = await getHash("./src/fixtures/text.md")
  expect(hash).toMatchSnapshot()
})

test("Encode WOFF", async () => {
  const hash = await getHash("./src/fixtures/font.woff")
  expect(hash).toMatchSnapshot()
})

test("Encode PNG", async () => {
  const hash = await getHash("./src/fixtures/image.png")
  expect(hash).toMatchSnapshot()
})

test("Encode PNG with null options", async () => {
  const hash = await getHash("./src/fixtures/image.png", null)
  expect(hash).toMatchSnapshot()
})

test("Encode text - Base26", async () => {
  const hash = await getHash("./src/fixtures/text.md", { encoding: 26 })
  expect(hash).toMatchSnapshot()
})

test("Encode WOFF - Base26 (alternative)", async () => {
  const hash = await getHash("./src/fixtures/font.woff", { encoding: "base26" })
  expect(hash).toMatchSnapshot()
})

test("Encode PNG - Base26", async () => {
  const hash = await getHash("./src/fixtures/image.png", { encoding: 26 })
  expect(hash).toMatchSnapshot()
})

test("Encode text - SHA256", async () => {
  const hash = await getHash("./src/fixtures/text.md", { algorithm: "sha256" })
  expect(hash).toMatchSnapshot()
})

test("Encode WOFF - SHA256", async () => {
  const hash = await getHash("./src/fixtures/font.woff", { algorithm: "sha256" })
  expect(hash).toMatchSnapshot()
})

test("Encode PNG - SHA256", async () => {
  const hash = await getHash("./src/fixtures/image.png", { algorithm: "sha256" })
  expect(hash).toMatchSnapshot()
})

test("Encode text - Blake3", async () => {
  const hash = await getHash("./src/fixtures/text.md", { algorithm: "blake3" })
  expect(hash).toMatchSnapshot()
})

test("Encode WOFF - Blake3", async () => {
  const hash = await getHash("./src/fixtures/font.woff", { algorithm: "blake3" })
  expect(hash).toMatchSnapshot()
})

test("Encode PNG - Blake3", async () => {
  const hash = await getHash("./src/fixtures/image.png", { algorithm: "blake3" })
  expect(hash).toMatchSnapshot()
})

test("FileName text", async () => {
  const hash = await getHashedName("./src/fixtures/text.md")
  expect(hash).toMatchSnapshot()
})

test("FileName WOFF", async () => {
  const hash = await getHashedName("./src/fixtures/font.woff")
  expect(hash).toMatchSnapshot()
})

test("FileName PNG", async () => {
  const hash = await getHashedName("./src/fixtures/image.png")
  expect(hash).toMatchSnapshot()
})

test("Encode text - with invalid hash crashes", async () => {
  let errorMessage

  try {
    await getHash("./src/fixtures/text.md", { algorithm: "other" })
  } catch (error) {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    errorMessage = error.message
  }

  expect(errorMessage).toMatchSnapshot()
})

test("Class: Defaults", async () => {
  const hasher = new Hasher()
  await hasher.init()
  hasher.update(testString)
  expect(hasher.digest()).toMatchSnapshot()
})

test("Class: SHA256", async() => {
  const hasher = new Hasher({ algorithm: "sha256" })
  await hasher.init()
  hasher.update(testString)
  expect(hasher.digest()).toMatchSnapshot()
})

test("Class: Defaults - Base64", async() => {
  const hasher = new Hasher({ encoding: "base64" })
  await hasher.init()
  hasher.update(testString)
  expect(hasher.digest()).toMatchSnapshot()
})

test("Class: SHA256 - Base64", async() => {
  const hasher = new Hasher({ algorithm: "sha256", encoding: "base64" })
  await hasher.init()
  hasher.update(testString)
  expect(hasher.digest()).toMatchSnapshot()
})

test("Class: Defaults - Hex", async() => {
  const hasher = new Hasher({ encoding: "hex" })
  await hasher.init()
  hasher.update(testString)
  expect(hasher.digest()).toMatchSnapshot()
})

test("Class: SHA256 - Hex", async() => {
  const hasher = new Hasher({ algorithm: "sha256", encoding: "hex" })
  await hasher.init()
  hasher.update(testString)
  expect(hasher.digest()).toMatchSnapshot()
})
