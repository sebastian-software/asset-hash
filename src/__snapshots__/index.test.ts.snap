// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class: Defaults - Base64 1`] = `"acqKwWZJNjc="`;

exports[`Class: Defaults - Hex 1`] = `"69ca8ac166493637"`;

exports[`Class: Defaults 1`] = `"BBhBKYGUhIr"`;

exports[`Class: SHA256 - Base64 1`] = `"A9yvJrOLcTDHCHQMineSc86MSOLP1obgCA/mxE9daiE="`;

exports[`Class: SHA256 - Hex 1`] = `"03dcaf26b38b7130c708740c8a779273ce8c48e2cfd686e0080fe6c44f5d6a21"`;

exports[`Class: SHA256 1`] = `"eMjdHMljxndqZBeKseUgirIJpLEAJzkJYPqzSEUdaXygl"`;

exports[`Encode PNG - Base26 1`] = `"ckrbqwya"`;

exports[`Encode PNG - Farmhash 1`] = `"SAKCwDbT"`;

exports[`Encode PNG - SHA256 1`] = `"TbJLQQkh"`;

exports[`Encode PNG 1`] = `"EJVIlSMa"`;

exports[`Encode PNG with null options 1`] = `"EJVIlSMa"`;

exports[`Encode WOFF - Base26 (alternative) 1`] = `"tmwfjtfl"`;

exports[`Encode WOFF - Farmhash 1`] = `"iTevvBHn"`;

exports[`Encode WOFF - SHA256 1`] = `"qzUDgbQl"`;

exports[`Encode WOFF 1`] = `"jpmdEypL"`;

exports[`Encode text - Base26 1`] = `"fkknifgr"`;

exports[`Encode text - Farmhash 1`] = `"KtZHMeFL"`;

exports[`Encode text - SHA256 1`] = `"gxuQSKsO"`;

exports[`Encode text 1`] = `"eVKeeoqr"`;

exports[`FileName PNG 1`] = `"EJVIlSMa.png"`;

exports[`FileName WOFF 1`] = `"jpmdEypL.woff"`;

exports[`FileName text 1`] = `"eVKeeoqr.md"`;
